pkg/plugin/linuxutil/netstat_stats_linux.go

```diff
@@ -271,12 +274,13 @@ func (nr *NetstatReader) updateMetrics() {
 }
 
 func validateRemoteAddr(addr string) bool {
+	nodeIP := os.Getenv("NODE_IP")
 	if addr == "" {
 		return false
 	}
 
 	// ignore localhost addresses.
-	if strings.Contains(addr, "127.0.0") {
+	if strings.Contains(addr, "127.0.0") || addr == nodeIP {
```

Comment:
Any other possibilities? Like `0.0.0.0` or `localhost`?

Reply:
This should also not be `strings.Contains`. This will match `8.127.0.0`, for example. `strings.HasPrefix` is more appropriate (though most appropriate is using the actual IP types in `net`).

Reply:
Nice catch, I updated it to use HasPrefix.